packages/dataviews/src/components/dataviews/index.tsx

```diff
@@ -47,11 +47,13 @@ type DataViewsProps< Item > = {
 	onClickItem?: ( item: Item ) => void;
 	isItemClickable?: ( item: Item ) => boolean;
 	header?: ReactNode;
+	getItemLevel?: ( item: Item ) => number;
 } & ( Item extends ItemWithId
 	? { getItemId?: ( item: Item ) => string }
 	: { getItemId: ( item: Item ) => string } );
 
 const defaultGetItemId = ( item: ItemWithId ) => item.id;
+const defaultGetItemLevel = ( item: any ) => item.level;
```

Comment:
I think we shouldn't provide any default here. (By default, dataviews shouldn't support hierarchy) and hierarchy behavior should only trigger if this is defined and the view flag is defined too.

Reply:
Removed the default.

I think providing the default is a nice touch for consumers, and they should only care about settings `view.showLevels`. Said this, I'm fine to start without a default and re-evaluate later.

Reply:
What I don't like about this is that it gives meaning to `level` property. I know we already do this for `id` but I don't like too many reserved words.

Reply:
Yeah, I see your point, I wasn't super excited about this approach either. What would be a better alternative?